modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/display/context/ObjectEntryDisplayContextImpl.java

```diff
@@ -183,18 +183,22 @@ public ObjectEntryDisplayContextImpl(
 	}
 
 	public String getAPIURL() throws PortalException {
-		ObjectRelationship objectRelationship =
-			_objectRelationshipLocalService.fetchObjectRelationship(
-				ParamUtil.getLong(
-					_objectRequestHelper.getRequest(), "objectRelationshipId"));
-
 		String externalReferenceCode = null;
 
 		if (_objectEntry != null) {
 			externalReferenceCode = _objectEntry.getExternalReferenceCode();
 		}
 
+		ObjectRelationship objectRelationship =
+			_objectRelationshipLocalService.fetchObjectRelationship(
+				ParamUtil.getLong(
+					_objectRequestHelper.getRequest(), "objectRelationshipId"));
+
 		if ((objectRelationship == null) || !objectRelationship.isEdge()) {
+			externalReferenceCode =
+				(externalReferenceCode != null) ? externalReferenceCode :
+					_getObjectEntry().getExternalReferenceCode();
+
```

Comment:
I think it looks cleaner and easier to follow with this structure:

```
ObjectRelationship objectRelationship =
		_objectRelationshipLocalService.fetchObjectRelationship(
			ParamUtil.getLong(
				_objectRequestHelper.getRequest(), "objectRelationshipId"));

	String externalReferenceCode = null;
	
	ObjectEntry objectEntry = _getObjectEntry();

	if (objectEntry != null) {
		externalReferenceCode = objectEntry.getExternalReferenceCode();
	}

	if ((objectRelationship == null) || !objectRelationship.isEdge()) {
		return _getAPIURL(externalReferenceCode, getObjectDefinition1());
	}
```

This way, the flow is more straightforward.